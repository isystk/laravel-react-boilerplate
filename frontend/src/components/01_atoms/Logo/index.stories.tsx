import { Meta, Story } from '@storybook/react'
import React from 'react'
import Logo, { LogoProps } from './index'

export default {
  title: '01_atoms/Logo',
  component: Logo,
} as Meta

const Template: Story = () => {
  const props: LogoProps = {
    name: 'sample',
    link: '#',
  }
  return <Logo {...props} />
}

export const Primary = Template.bind({})
Primary.storyName = 'プライマリ'
Primary.args = {}
