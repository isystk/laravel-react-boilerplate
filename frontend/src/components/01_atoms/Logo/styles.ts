import { css, keyframes } from '@emotion/css'

export const logo = css``
