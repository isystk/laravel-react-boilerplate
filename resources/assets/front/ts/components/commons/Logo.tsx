import React, { FC } from "react";
import { Link } from "react-router-dom";
import { Url } from "@/constants/url";

export const Logo: FC = () => {
    return (
        <Link className="flex items-center" to={Url.TOP}>
            <img src="/assets/front/image/logo.png" alt="" className="h-10 md:h-auto" />
        </Link>
    );
};

export default Logo;
