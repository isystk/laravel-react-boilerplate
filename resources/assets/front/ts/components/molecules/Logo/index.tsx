import './styles.module.scss';
import Image from '@/components/atoms/Image';
import {APP_NAME} from "@/constants/app";
import {Link} from "react-router-dom";
import { Url } from "@/constants/url";

export type Props = object

const Logo = () => {
    return (
        <Link
            to={Url.TOP}
            className={`flex items-center`}
        >
            <Image src="/assets/front/image/logo.png" width={200} height={60} alt={APP_NAME} className="h-10 md:h-auto" />
        </Link>
    );
};

export default Logo;

