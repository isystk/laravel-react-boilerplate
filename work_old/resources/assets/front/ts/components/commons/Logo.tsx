import React, { FC } from "react";
import { Link } from "react-router-dom";
import { Url } from "@/constants/url";

export const Logo: FC = () => {
    return (
        <Link className="header_logo" to={Url.TOP}>
            <img src="/assets/front/image/logo.png" alt="" />
        </Link>
    );
};

export default Logo;
