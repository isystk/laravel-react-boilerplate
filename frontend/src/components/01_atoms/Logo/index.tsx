import React, { FC } from 'react'
import { ContainerProps, WithChildren } from 'types'
import { connect } from '@/components/hoc'
import * as styles from './styles'
import { APP_NAME } from '@/constants'
import { Url } from '@/constants/url'
import Image, {
  PresenterProps as ImagePresenterProps,
} from '@/components/01_atoms/Image'
import { useRouter } from 'next/router'

/** LogoProps Props */
export type LogoProps = {
  name?: string
  link?: string
}
/** Presenter Props */
export type PresenterProps = LogoProps & {
  router
}

/** Presenter Component */
const LogoPresenter: FC<PresenterProps> = ({
  name = APP_NAME,
  link = Url.Top,
  router,
}) => {
  const props: ImagePresenterProps = {
    src: '/images/logo.png',
    alt: name,
    className: 'w-20 md:w-40',
    lazy: false,
  }
  return (
    <>
      <a
        href="#"
        className={styles.logo}
        onClick={(e) => {
          e.preventDefault()
          router.push(link)
        }}
      >
        <Image {...props} />
      </a>
    </>
  )
}

/** Container Component */
const LogoContainer: React.FC<ContainerProps<LogoProps, PresenterProps>> = ({
  presenter,
  ...props
}) => {
  const router = useRouter()
  return presenter({
    router,
    ...props,
  })
}

export default connect<LogoProps, PresenterProps>(
  'Logo',
  LogoPresenter,
  LogoContainer
)
