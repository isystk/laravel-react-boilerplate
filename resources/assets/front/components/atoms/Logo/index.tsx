import styles from './styles.module.scss';
import Image from '@/components/atoms/Image';
import { Link } from 'react-router-dom';
import { Url } from '@/constants/url';
import Env from '@/constants/env';
import logoImage from '@/assets/images/logo.png';

export type Props = {
  hasLink?: boolean;
};

const Logo = ({ hasLink = true }: Props) => {
  return hasLink ? <LinkLogo /> : <NoLinkLogo />;
};

const LinkLogo = () => {
  return (
    <Link to={Url.top} className={`flex items-center`}>
      <Image
        src={logoImage as string}
        width={200}
        height={60}
        alt={Env.appName}
        className={styles.logoImage}
      />
    </Link>
  );
};

const NoLinkLogo = () => {
  return (
    <Image
      src={logoImage as string}
      width={200}
      height={60}
      alt={Env.appName}
      className={styles.logoImage}
    />
  );
};

export default Logo;
