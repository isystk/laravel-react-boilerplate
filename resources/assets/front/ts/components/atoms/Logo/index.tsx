import styles from './styles.module.scss';
import Image from '@/components/atoms/Image';
import {Link} from "react-router-dom";
import { Url } from "@/constants/url";
import Env from "@/constants/env";

export type Props = {
    hasLink?: boolean;
}

const Logo = ({hasLink = true}: Props) => {
    return (
        hasLink ? <LinkLogo /> : <NoLinkLogo />
    );
};

const LinkLogo = () => {
    return (
        <Link
            to={Url.top}
            className={`flex items-center`}
        >
            <Image src="/assets/front/image/logo.png" width={200} height={60} alt={Env.appName} className={styles.logoImage} />
        </Link>
    );
};

const NoLinkLogo = () => {
    return (
        <Image src="/assets/front/image/logo.png" width={200} height={60} alt={Env.appName} className={styles.logoImage} />
    );
};

export default Logo;

