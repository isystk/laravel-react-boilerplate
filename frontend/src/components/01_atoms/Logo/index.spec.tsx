import React from 'react'
import { render } from '@testing-library/react'
import Logo, { LogoProps } from './index'

describe('LandingPageTemplate', () => {
  it('Match Snapshot', () => {
    const props: LogoProps = {
      name: 'sample',
      link: '#',
    }
    const { asFragment } = render(<Logo {...props} />)
    expect(asFragment()).toMatchSnapshot()
  })
})
