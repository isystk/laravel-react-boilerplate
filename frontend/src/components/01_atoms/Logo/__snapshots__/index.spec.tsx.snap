// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LandingPageTemplate Match Snapshot 1`] = `
<DocumentFragment>
  <a
    class="css-0"
    href="#"
  >
    <div
      class="w-20 md:w-40"
    >
      <img
        alt="sample"
        class="css-s08p0c "
        src="/images/logo.png"
      />
    </div>
  </a>
</DocumentFragment>
`;
